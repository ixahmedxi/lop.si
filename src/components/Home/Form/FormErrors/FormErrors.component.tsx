import { FiAlertTriangle } from 'react-icons/fi'
import { SxStyleProp, Text } from 'theme-ui'

interface PropTypes {
  errors: string | null
}

const styles: SxStyleProp = {
  text: {
    color: '#e8505b',
    pl: [4, 4, 6],
    pt: 3,
    display: 'flex',
    alignItems: 'center',
    fontSize: [0, 0, 1],
    svg: {
      mr: 1
    }
  }
}

export const FormErrors = ({ errors }: PropTypes) => {
  if (errors === null) {
    return null
  }

  return (
    <Text as="p" sx={styles.text}>
      <FiAlertTriangle />
      {errors}
    </Text>
  )
}
