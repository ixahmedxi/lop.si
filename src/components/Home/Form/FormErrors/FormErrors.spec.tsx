import { render } from 'test-utils'
import { FormErrors } from './FormErrors.component'

describe('FormErrors component', () => {
  it('matches snapshot', () => {
    const { container } = render(<FormErrors errors="form error" />)
    expect(container).toMatchSnapshot()
  })

  it('renders null when errors are null', () => {
    const { container } = render(<FormErrors errors={null} />)
    expect(container).toBeEmptyDOMElement()
  })

  it('renders the error string when passed', () => {
    const { getByText } = render(<FormErrors errors="form error" />)
    getByText('form error')
  })
})
