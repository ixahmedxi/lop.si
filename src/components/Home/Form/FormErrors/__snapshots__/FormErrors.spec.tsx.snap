// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrors component matches snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: #e8505b;
  padding-left: 32px;
  padding-top: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
}

@media screen and (min-width:576px) {
  .emotion-0 {
    padding-left: 32px;
    font-size: 12px;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    padding-left: 48px;
    font-size: 16px;
  }
}

.emotion-0 svg {
  margin-right: 8px;
}

<div>
  <p
    class="emotion-0"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
      />
      <line
        x1="12"
        x2="12"
        y1="9"
        y2="13"
      />
      <line
        x1="12"
        x2="12"
        y1="17"
        y2="17"
      />
    </svg>
    form error
  </p>
</div>
`;
